src/commands/trigger.cpp

```diff
@@ -0,0 +1,728 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// trigger.cpp
+//
+// Identification: src/commands/trigger.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+
+#include "commands/trigger.h"
+#include "parser/pg_trigger.h"
+#include "common/logger.h"
+#include "expression/constant_value_expression.h"
+#include "catalog/column_catalog.h"
+#include "type/type.h"
+#include "type/value.h"
+#include "type/value_factory.h"
+
+namespace peloton {
+namespace commands {
+Trigger::Trigger(const peloton::planner::CreatePlan &plan) {
+  trigger_name = plan.GetTriggerName();
+  trigger_funcname = plan.GetTriggerFuncName();
+  trigger_args = plan.GetTriggerArgs();
+  trigger_columns = plan.GetTriggerColumns();
+  trigger_when = plan.GetTriggerWhen();
+  trigger_type = plan.GetTriggerType();
+}
+
+//===--------------------------------------------------------------------===//
+// DEPRECATED
+//===--------------------------------------------------------------------===//
+Trigger::Trigger(std::string name,
+                 UNUSED_ATTRIBUTE std::string function_name,
+                 UNUSED_ATTRIBUTE std::string arguments,
+                 UNUSED_ATTRIBUTE std::string fire_condition) {
+  trigger_name = name;
+  // to be continue...
+
+}
+Trigger::Trigger(std::string name,
+                 int16_t type,
+                 std::string function_name,
+                 std::string arguments,
+                 std::string fire_condition) {
+  trigger_name = name;
+  std::vector<std::string> strs;
+  boost::split(strs, function_name, boost::is_any_of(","));
+  for (unsigned int i = 0; i < strs.size(); i++) {
+    trigger_funcname.push_back(strs[i]);
+  }
+  strs.clear();
+  boost::split(strs, arguments, boost::is_any_of(","));
+  for (unsigned int i = 0; i < strs.size(); i++) {
+    trigger_args.push_back(strs[i]);
+  }
+  trigger_type = type;
+  trigger_when = DeserializeWhen(fire_condition);
+}
+
+std::string Trigger::SerializeWhen(oid_t table_oid, concurrency::Transaction *txn) {
```

Comment:
This seems really gross and different from how we serialize other things. It also seems really error prone. We will have to revisit this.